### DISCUSSION TITLE
Building with Recent Version of MLIR

### DISCUSSION THREAD

**Discussion by aferr** [aqjune/mlir-tv#354] [General]
Hello,

I am currently trying to build MLIR-TV with a fairly recent version of LLVM/LL (the git commit date is April 6th).

I get quite a few errors about differences between what MLIR-TV expects from the MLIR headers and my current headers. For example, the "Arithmetic" dialect headers have been renamed to just "Arith" and some accessors of GeneircOp, have been renamed (for example type() is now getType() ).

I am currently patching this on my machine, but I wonder if it is easier if:
 - there is some version of MLIR I should be using to build MLIR-TV. What version is this?
 - you are already working on updating MLIR-TV to address this.
Thanks

**Comment by aqjune** [comment]
Hello @aferr,

MLIR-TV is supposed to work for the latest MLIR, but I have been occupied with some other works for a while.
Patches for mlir-tv are always welcome :)

There has been https://github.com/aqjune/mlir-tv/pull/352 for patching MLIR-TV to work on the latest MLIR.
@Hatsunespica do you mind if I ask whether the work is still on progress, or would it be okay if aferr works on this instead..?

**Comment by aqjune** [comment]
Hi, we are not getting a reply from @Hatsunepica.
If you have already written a patch that supports the latest MLIR and would like to share it, please feel free to make a pull request!

**Comment by aqjune** [comment]
I slightly updated the existing PR to support the latest MLIR and merged it - now it should work!
